compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposePanel.desktop.kt

```diff
@@ -171,31 +171,48 @@ class ComposePanel @ExperimentalComposeUiApi constructor(
     }
 
     override fun add(component: Component): Component {
-        if (bridge == null) {
-            return component
-        }
-        val clipComponent = ClipComponent(component)
-        clipMap[component] = clipComponent
-        bridge!!.clipComponents.add(clipComponent)
-        return super.add(component, Integer.valueOf(0))
+        addToLayer(component, componentLayer)
+        return component
+    }
+
+    private fun addBridge(bridge: ComposeBridge) {
+        addToLayer(bridge.invisibleComponent, bridgeLayer)
+        addToLayer(bridge.component, bridgeLayer)
     }
 
-    override fun remove(component: Component) {
-        bridge!!.clipComponents.remove(clipMap[component]!!)
-        clipMap.remove(component)
-        super.remove(component)
+    private fun addToLayer(component: Component, layer: Int) {
+        if (renderApi == GraphicsApi.METAL && bridge !is SwingComposeBridge) {
+            // Applying layer on macOS makes our bridge non-transparent
+            // But it draws always on top, so we can just add it as-is
+            // TODO: Figure out why it makes difference in transparency
+            super.add(component, 0)
+        } else {
+            super.setLayer(component, layer)
+            super.add(component)
+        }
     }
 
+    private val bridgeLayer: Int = 10
+
+    // Place it to top if alpha blending doesn't work
+    private val componentLayer: Int
+        get() = if (isAlphaBlendingWorking) 0 else 20
+
+    // TODO: Support for all platforms
+    private val isAlphaBlendingWorking
+        get() = bridge is SwingComposeBridge
+            || renderApi == GraphicsApi.METAL // || renderApi == GraphicsApi.DIRECT3D
+
     override fun addNotify() {
         super.addNotify()
 
         // After [super.addNotify] is called we can safely initialize the bridge and composable
         // content.
-        if (bridge == null) {
-            bridge = createComposeBridge()
+        if (this.bridge == null) {
+            val bridge = createComposeBridge()
+            this.bridge = bridge
             initContent()
-            super.add(bridge!!.invisibleComponent, Integer.valueOf(1))
-            super.add(bridge!!.component, Integer.valueOf(1))
+            addBridge(bridge)
```

Comment:
`addBridge` is only called once. I'd at least put it right next to this function (or even make it a local function). But really, I would just inline it.

Reply:
Done